src/core/value.js

```diff
@@ -0,0 +1,134 @@
+/* @flow */
+
+'use strict';
+
+const GlobalBigNumber = require('bignumber.js');
+
+const BigNumber = GlobalBigNumber.another({
+  ROUNDING_MODE: GlobalBigNumber.ROUND_HALF_UP,
+  DECIMAL_PLACES: 40
+});
+
+class Value {
```

Comment:
We should make this an abstract base class so that there are only two Value classes that appear in the code: in the constructor, throw an exception if `new.target === Value`. Also, in all the places that instantiate Value directly, replace it with either XRPValue or IOUValue.